pkg/providers/aws/cluster_network_provider_test.go

```diff
@@ -1061,6 +1061,29 @@ func TestNetworkProvider_CreateNetwork(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "verify vpc limit exceeded returns an error",
+			fields: fields{
+				Client: fake.NewFakeClientWithScheme(scheme, buildTestInfra()),
+				RdsApi: buildMockRdsClient(nil),
+				Ec2Api: buildMockEc2Client(func(ec2Client *mockEc2Client) {
+					ec2Client.vpcs = buildValidClusterVPC(defaultNonOverlappingCidr)
+					ec2Client.createVpcFn = func(input *ec2.CreateVpcInput) (*ec2.CreateVpcOutput, error) {
+						return &ec2.CreateVpcOutput{}, awserr.New("VpcLimitExceeded", "The maximum number of VPCs has been reached.", nil)
+					}
+					ec2Client.describeRouteTablesFn = func(input *ec2.DescribeRouteTablesInput) (*ec2.DescribeRouteTablesOutput, error) {
+						return &ec2.DescribeRouteTablesOutput{RouteTables: []*ec2.RouteTable{}}, nil
+					}
+				}),
+				ElasticacheApi: buildMockElasticacheClient(nil),
+				Logger:         logrus.NewEntry(logrus.StandardLogger()),
+			},
+			args: args{
+				ctx:  context.TODO(),
+				CIDR: buildValidCIDR(validCIDRSixteen),
+			},
+			wantErr: true,
+		},
```

Comment:
Hm, I remember that it is nearly impossible to reach the `InvalidVpcRange` error code here, but still - do we want to have a unit test for it? I mean CRO is famous for the unit test coverage :thinking:

Reply:
Hm, when running that unit test I got him passing, but the error I've got was `failed to get cluster vpc`. Shouldn't it be `VpcLimitExceeded`? :thinking: 
It had an issue with `failed to get cluster vpc id, no vpc found with osd cluster tag: could not find cluster associated subnets with clusterID test` and I remember messing around this [here](https://github.com/integr8ly/cloud-resource-operator/pull/244) and, probably, the way I was fixing unit tests will work as well :thinking:

Reply:
Good spot, I should've been creating the subnets in the test. I've updated in latest commit

Reply:
>Hm, I remember that it is nearly impossible to reach the InvalidVpcRange error code here, but still - do we want to have a unit test for it? I mean CRO is famous for the unit test coverage thinking

I could add an additional test for the `InvalidVpcRange` error also - as you say though I don't think that AWS error is reachable as it should be caught further up

Reply:
Hmm, that's a tricky one. Right now this error is unreachable, yes, but we still could care about it because in the future it might change (the catch of this error will be removed and the person will see that we catch it here) or something mysterious might happen and previous catch won't work. Personally, for me, it is better to be safe than sorry. Now the philosophical one: if we decided to have a piece of code, even it is unreachable now, and could have a test for it - shouldn't we? On one hand, we have potentially wasted development time and already way too chunky test file, on the other hand, it won't hurt to have en extra coverage? :thinking:

Reply:
Sure, I agree. I'll add the test as a fall through in case the other check is removed in favour of the AWS errors.